### PR TITLE
FOUNDATIONS: Modify Language

### PR DESCRIPTION
**Pull request by imanvt**

### CODE REVIEW

File: Sofee.Core.Api.Tests.Unit/Services/Foundations/Languages/LanguageServiceTests.Logic.Modify.cs [hunk 1]
```diff
@@ -0,0 +1,83 @@
+﻿using FluentAssertions;
```
Comment (ShriHumrudha) [inline Sofee.Core.Api.Tests.Unit/Services/Foundations/Languages/LanguageServiceTests.Logic.Modify.cs:1]: Add copyrights header. Run code cleanupl

File: Sofee.Core.Api.Tests.Unit/Services/Foundations/Languages/LanguageServiceTests.Logic.Modify.cs [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+                GetRandomDateTimeOffset();
+
+            Language randomLanguage =
+                CreateRandomLanguage(randomDate);
+
+            Language inputLanguage =
+                randomLanguage;
+
+            inputLanguage.UpdatedDate =
+                randomDate.AddMinutes(1);
```
Comment (ShriHumrudha) [inline Sofee.Core.Api.Tests.Unit/Services/Foundations/Languages/LanguageServiceTests.Logic.Modify.cs:27]: Use a random number to add the days than 1.

File: Sofee.Core.Api.Tests.Unit/Services/Foundations/Languages/LanguageServiceTests.Logic.Modify.cs [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+namespace Sofee.Core.Api.Tests.Unit.Services.Foundations.Languages
+{
+    public partial class LanguageServiceTests
+    {
+        [Fact]
+        public async Task ShouldModifyLanguageAsync()
+        {
+            // given
+            DateTimeOffset randomDate =
```
Comment (ShriHumrudha) [inline Sofee.Core.Api.Tests.Unit/Services/Foundations/Languages/LanguageServiceTests.Logic.Modify.cs:17]: Break the line if it is beyond 120 or belly pattern. I do not see a need here.

File: Sofee.Core.Api.Tests.Unit/Services/Foundations/Languages/LanguageServiceTests.Logic.Modify.cs [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+                        .ReturnsAsync(storageLanguage);
+
+            this.storageBrokerMock.Setup(broker =>
+                broker.UpdateLanguageAsync(
+                    inputLanguage))
+                        .ReturnsAsync(updatedLanguage);
+
+            // when
+            Language actualLanguage =
+                await this.languageService.
```
Comment (ShriHumrudha) [inline Sofee.Core.Api.Tests.Unit/Services/Foundations/Languages/LanguageServiceTests.Logic.Modify.cs:57]: . should be flown to next line

File: Sofee.Core.Api/Services/Foundations/Languages/LanguageService.Validations.cs [hunk 5]
```diff
@@ -32,6 +32,28 @@ private void ValidateLanguageOnAdd(Language language)
                 Parameter: nameof(Language.UpdatedDate)));
         }
 
+        private void ValidateLanguageOnModify(Language language)
```
Comment (ShriHumrudha) [inline Sofee.Core.Api/Services/Foundations/Languages/LanguageService.Validations.cs:35]: We need a failing test before this.

**Review by ShriHumrudha — CHANGES_REQUESTED** [review]

File: Sofee.Core.Api/Services/Foundations/Languages/LanguageService.Validations.cs (hunk 5 above)
Comment (imanvt) [reply Sofee.Core.Api/Services/Foundations/Languages/LanguageService.Validations.cs:35]: I messed up the ordering, will do it properly in future PRs.

**Pull request closed without merging**